features/FEATURE_LWIP/lwip-interface/EthernetInterface.cpp

```diff
@@ -46,11 +51,12 @@ nsapi_error_t EthernetInterface::set_dhcp(bool dhcp)
 
 nsapi_error_t EthernetInterface::connect()
 {
-    return mbed_lwip_bringup_2(_dhcp, false,
+    return mbed_lwip_bringup_3(_dhcp, false,
             _ip_address[0] ? _ip_address : 0,
             _netmask[0] ? _netmask : 0,
             _gateway[0] ? _gateway : 0,
-            DEFAULT_STACK);
+            DEFAULT_STACK, 
+            netif_status_irq);
```

Comment:
How about adding a `mbed_lwip_attach` call, rather than another parameter to the bringup? That would align better with the other APIs, I think, and avoid adding this "bringup_3".